google/cloud/security/scanner/audit/gce_networks_rules_engine.py

```diff
@@ -0,0 +1,233 @@
+        #    print(i)
+        #   print(rule)
+        #  self.add_rule(rule, i)
+
+    def add_rule(self, rule_def, rule_index):
+        """Add a rule to the rule book.
+
+        Args:
+            rule_def: A dictionary containing rule definition properties.
+            rule_index: The index of the rule from the rule definitions.
+            Assigned automatically when the rule book is built.
+        """
+        print("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
+        print(rule_def)
+        print(type(rule_def))
+        print(rule_index)
+        print("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
+
+        project = rule_def.get('project')
+        network = rule_def.get('network')
+        is_external_network = rule_def.get('is_external_network')
+        enforced_networks = rule_def.get('whitelist')
+
+        if (project is None) or (network is None) or (enforced_networks is None):
+            raise audit_errors.InvalidRulesSchemaError('Faulty rule {}'.format(rule_def.get('name')))
+
+#TODO make ENforcedNetworks type in gcp_type
+#TODO maybe some json magic on globified stuff
+        rule_def_resource = gn.GceNetwork(
+            escape_and_globify(project),
+            escape_and_globify(network),
+            is_external_network,
+            enforced_networks)
+
+        rule = Rule(rule_name=rule_def.get('name'), rule_index=rule_index, rules=rule_def_resource)
+
+        resource_rules = self.resource_rules_map.get(rule_index)
+
+        if not resource_rules:
+            self.resource_rules_map[rule_index] = rule
+
+    def get_resource_rules(self):
+        """Get all the resource rules for (resource, RuleAppliesTo.*).
+
+        Args:
+            resource: The resource to find in the ResourceRules map.
+
+        Returns:
+            A list of ResourceRules.
+        """
+        resource_rules = []
+
+        for resource_rule in self.resource_rules_map:
+            resource_rules.append(self.resource_rules_map[resource_rule])
+
+        return resource_rules
+
+
+
+
+class Rule(object):
+    """Rule properties from the rule definition file.
+    Also finds violations.
+    """
+
+    def __init__(self, rule_name, rule_index, rules):
+        """Initialize.
+
+        Args:
+            rule_name: Name of the loaded rule
+            rule_index: The index of the rule from the  definitions
+            rules: the gce instance 
+        """
+        self.rule_name = rule_name
+        self.rule_index = rule_index
+        self.rules = rules
+#TODO rename enforced_networks_rules?
+    def find_policy_violations(self, network_interface):
+        """
+        enforced_networks string of the name of a network 
+
+        """
+
+        if self.rules.project != '^.+$':
+            project_name_bool = re.match(self.rules.project, network_interface.project)
+
+        if self.rules.network != '^.+$':
+            network_name_bool = re.match(self.rules.network,
```

Comment:
I hope I am not misunderstanding something from our chat, but should this be checking against the ```whitelist``` in the rule?